bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -161,17 +163,39 @@ with h5py.File(args.template_file, "r") as f_src:
 temp_samples = numpy.vstack((template_pars)).T
 
 
+# Rejection Sampling
+if 'mchirp' not in args.fit_param:
+    raise ValueError("mchirp does not exist in args.fit_param")
+mchirp_index = args.fit_param.index('mchirp')
+if args.mchirp_downsample_power is not None:
+    power_vals = template_pars[mchirp_index] ** args.mchirp_downsample_power
+    f_dest.attrs['mchirp_downsample_power'] = args.mchirp_downsample_power
+    probabilities = power_vals / numpy.max(power_vals)
+    rand_nums = numpy.random.uniform(0, 1, len(mass1))
+    ind = rand_nums < probabilities
+    new_mchirp = template_pars[mchirp_index][ind]
+    additional_params = numpy.delete(temp_samples, mchirp_index, axis=1)[ind]
```

Comment:
here, is it necessary to treat mchirp separately from the other template parameters?